MAGIC variance reduction

kkiesling opened the pull request:
I added the MAGIC variance reduction technique to variancereduction.py. @makeclean, you should review the method. Tests were also added.

Theory manual addition still needed.

issue #344

scopatz commented:
Looks like this could use a rebase | merge from develop

scopatz commented:
Hey @kkiesling - now that this passes CI and is mergeable, is this ready for another review (and merge?)?

scopatz commented:
I think that this would be good to have for v0.5, if it can make it in there.

kkiesling commented:
If you are okay with merging it without the documentation, then I believe it is ready. I can't make any promises to have the documentation written until after March 1st unfortunately but it will be done in that week if I can't get it done earlier.

scopatz commented:
Hi @kkiesling - I think we need documentation.  Why don't we leave this in v0.5 for now, and if it happens to be the last remaining v0.5 issue open we'll make a decision then?

scopatz commented on pyne/variancereduction.py:
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
+
+    Parameters:
+    -----------
+        meshtally : a single PyNE mcnp.MeshTally obj
```
Unindent these lines please to followi the numpydoc rules.

scopatz commented on pyne/variancereduction.py:
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
+            elements where the relative error on flux exceeds the tolerance.
+    """
+    
+    tolerance = kwargs["tolerance"] if "tolerance" in kwargs else 0.5
```
this is better as 

```python
tolerace = kwargs.get('tolerance', 0.5)
```

and similarly below.

scopatz commented on pyne/variancereduction.py:
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
+    null_value = kwargs["null_value"] if "null_value" in kwargs else 0.0
+    
+    # Convert particle name to the recognized abbreviation
+    particle = (meshtally.particle[0].upper() + meshtally.particle[1:])
```
you probably want to do `meshtally.capitalize()` here.

scopatz commented:
couple of minor style issues @kkiesling, otherwise it loosk good to go.

kkiesling commented:
Pinging this. I fixed the style issues, finished the manual entry, and added the test that @makeclean was talking about a while back. This should be good to go officially when Andy approves.

scopatz commented:
Looks good to me, @makeclean?

makeclean commented:
Yup yup. Merging.
